OptimoveNotificationServiceExtension/Sources/OptimoveNotificationService.swift

```diff
@@ -1,217 +1,133 @@
 //  Copyright © 2022 Optimove. All rights reserved.
 
 import Foundation
+import OptimoveCore
 import UIKit
 import UserNotifications
 
-public class OptimoveNotificationService {
-    private static let syncBarrier = DispatchSemaphore(value: 0)
-
-    public class func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
-        let bestAttemptContent = (request.content.mutableCopy() as! UNMutableNotificationContent)
-        let userInfo = request.content.userInfo
+public enum OptimoveNotificationService {
+    enum Error: String, LocalizedError {
+        case noBestAttemptContent
+        case userInfoNotValid
+    }
 
-        if !validateUserInfo(userInfo: userInfo) {
-            return
+    public static func didReceive(
+        _ request: UNNotificationRequest,
+        withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void
+    ) {
+        Task {
+            do {
+                let bestAttemptContent = try await didReceive(request)
+                contentHandler(bestAttemptContent)
+            } catch {
+                assertionFailure(error.localizedDescription)
+                contentHandler(request.content)
```

Comment:
Previously, if validateUserInfo returned false, contentHandler wasn't called. Can this cause a problem in case of missing keys in: ["custom", "a", "k.message", "data"] (can this even happen?)

Reply:
It is ok to call `contentHandler` with unmodified content if we cannot handle it. This callback will be invoked with the unmodified content in any case after a system timeout.